src/components/SingleList.jsx

```diff
@@ -1,19 +1,42 @@
 import { useNavigate } from 'react-router-dom';
 
 import './SingleList.css';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrashCan } from '@fortawesome/free-regular-svg-icons';
+import { deleteSingleList } from '../api/firebase';
+import { useAuth } from '../api';
 
 export function SingleList({ name, path, setListPath }) {
+	const { user } = useAuth();
+
 	const navigate = useNavigate();
 
 	function handleClick() {
 		setListPath(path);
 		navigate('list');
 	}
-
+	async function deleteSelectedList() {
+		if (
+			window.confirm(
+				`Are you sure you want to delete the ${name} list? This action cannot be undone and all items in this list will be deleted.`,
+			)
+		) {
+			try {
+				const result = await deleteSingleList(user.email, path);
+				alert(result.message);
+				setListPath(null);
+			} catch (error) {
+				alert(`Failed to delete this list: ${error.message}`);
+			}
+		}
+	}
 	return (
 		<div className="SingleList">
 			<li>
 				<button onClick={handleClick}>{name}</button>
+				<button aria-label={`Delete ${name}`} onClick={deleteSelectedList}>
+					<FontAwesomeIcon icon={faTrashCan} /> Delete
```

Comment:
I also removed the `Delete` text in between the opening and closing tag. 

Our aria label makes the icon accessible, so I removed the text, to make a cleaner UI, and to be closer to our wireframe designs.

If you do this also, our delete button styling will be consistent :-).